cpu/kinetis_common/lpm_arch.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2014 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_kinetis_common_lpm
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the kernel's power management interface
+ *
+ * @author      Joakim Gebart <joakim.gebart@eistec.se>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "arch/lpm_arch.h"
+#include "kinetis_lpm.h"
+
+/*
+ * Counting semaphores for inhibiting undesired power modes.
+ */
+atomic_int_t kinetis_lpm_inhibit_stop_sema = {0};
+atomic_int_t kinetis_lpm_inhibit_vlps_sema = {0};
+atomic_int_t kinetis_lpm_inhibit_lls_sema  = {0};
+
+static inline void wait(void)
+{
+    /* Clear the SLEEPDEEP bit to make sure we go into WAIT (sleep) mode instead
```

Comment:
over-commented,  /* wait for outstanding writes and go into WAIT (sleep) mode */ is suffiecient
There are 6 lines of comments and 3 lines of plain code :-)